src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppInstanceStatus.java

```diff
@@ -163,21 +176,41 @@ private DeploymentState mapState() {
 			result.put("container.restartCount", "" + containerStatus.getRestartCount());
 			if (containerStatus.getLastState() != null && containerStatus.getLastState().getTerminated() != null) {
 				result.put("container.lastState.terminated.exitCode",
-					"" + containerStatus.getLastState().getTerminated().getExitCode());
+						"" + containerStatus.getLastState().getTerminated().getExitCode());
 				result.put("container.lastState.terminated.reason",
-					containerStatus.getLastState().getTerminated().getReason());
+						containerStatus.getLastState().getTerminated().getReason());
 			}
 			if (containerStatus.getState() != null && containerStatus.getState().getTerminated() != null) {
 				result.put("container.state.terminated.exitCode",
-					"" + containerStatus.getState().getTerminated().getExitCode());
+						"" + containerStatus.getState().getTerminated().getExitCode());
 				result.put("container.state.terminated.reason", containerStatus.getState().getTerminated().getReason());
 			}
 		}
 		return result;
 	}
-}
-
-
 
+	private String nullSafe(String value) {
+		return value == null ? "" : value;
+	}
 
+	private String determineActuatorPathFromLivenessProbe(Pod pod) {
+		return pod.getSpec().getContainers().stream()
+				.filter((Container container) -> container.getLivenessProbe() != null &&
+						container.getLivenessProbe().getHttpGet() != null)
+				.findFirst()
+				.map(container ->
+						Paths.get(container.getLivenessProbe().getHttpGet().getPath()).getParent().toString())
+				.orElse("/actuator");
+	}
 
+	private String determineActuatorPortFromLivenessProbe(Pod pod, String path) {
+		IntOrString intOrString = pod.getSpec().getContainers().stream()
+				.filter((Container container) -> container.getLivenessProbe() != null &&
+						container.getLivenessProbe().getHttpGet() != null &&
+						container.getLivenessProbe().getHttpGet().getPath().equals(path))
+				.findFirst()
+				.map(container -> container.getLivenessProbe().getHttpGet().getPort())
+				.orElse(new IntOrString(8080));
+		return intOrString.getIntVal() != null ? String.valueOf(intOrString.getIntVal()) : intOrString.getStrVal();
```

Comment:
Wow! `IntOrString` is a terrible construct - I would expect at least some convenience API built-in such as `asStrVal` which did this for us. I checked - it is not there.

Reply:
Agree. I hit an NPE assuming I could get it either way.